lib/app-common.ts

```diff
@@ -18,12 +18,7 @@
 class CliSettings {
 	public readonly settings: any;
 	constructor() {
-		// TODO figure out why the typescript compiler attempts to type-check
-		// the `balena-settings-client` module (and then fails with errors) if
-		// a straighforward `require('balena-settings-client')` statement is
-		// used here. It may even be a compiler bug, because `tsconfig.json`
-		// has a `"skipLibCheck": true` setting.
-		this.settings = require(['balena', 'settings', 'client'].join('-'));
+		this.settings = require('balena-settings-client') as typeof import('balena-settings-client');
```

Comment:
The reason typescript had issues before was balena-settings-client was using the `.ts` files for types and not generating declaration files, it was an easy fix with https://github.com/balena-io-modules/balena-settings-client/pull/60/commits/4f3e00e719e26818908f67127f2113f4fcdd0d21